10주차/20056/20056_cpp_이지영.cpp

```diff
@@ -0,0 +1,136 @@
+#include <iostream>
+//#include <cstring>
+#include <algorithm>
+#include <vector>
+#include <limits.h>
+#include<cstring>
+using namespace std;
+
+struct F
+{
+    int x;
+    int y;
+    int Massive;
+    int Speed;
+    int dir;
+    //fireball ���ü
+};
+
+int dx[] = { -1, -1, 0, 1, 1, 1, 0, -1 };
+int dy[] = { 0, 1, 1, 1, 0, -1, -1, -1 };
+int td[] = { 0, 2, 4, 6 }; //��� Ȧ���̰ų� ¦��
+int fd[] = { 1, 3, 5 ,7 };
+
+int N, M, K;
+vector<F> mapp[51][51];
+vector<F> fireball;
+
+void Move()
+{
+    for (int i = 1; i <= N; i++)
+    {
+        for (int j = 1; j <= N; j++)
+        {
+            mapp[i][j].clear();
+        }
+    }
+
+    for (int i = 0; i < fireball.size(); i++)
+    {
+        int x = fireball[i].x;
+        int y = fireball[i].y;
+        int Mass = fireball[i].Massive;
+        int Speed = fireball[i].Speed;
+        int dir = fireball[i].dir;
+
+        int Move = Speed % N;
+        int nx = x + dx[dir] * Move;
+        int ny = y + dy[dir] * Move;
```

Comment:
나머지는 % 연산자를 활용하면 편해요!
```suggestion
        int ny = (y + dy[dir] * Move + N) % N;
```